spec/unit/MatrixAction.spec.js

```diff
@@ -0,0 +1,57 @@
+"use strict";
+const MatrixAction = require("../../lib/models/MatrixAction");
+
+describe("MatrixAction", function() {
+    it("should not highlight mentions to text without mentions", () => {
+        let action = new MatrixAction("message", "Some text", "Some text", null);
+        action.formatMentions({
+            "Some Person": "@foobar:localhost"
+        });
+        expect(action.text).toEqual("Some text");
+    });
+    it("should highlight a user", () => {
+        let action = new MatrixAction("message", "JCDenton, it's a bomb!", "JCDenton, it's a bomb!", null);
+        action.formatMentions({
+            "JCDenton": "@jc.denton:unatco.gov"
+        });
+        expect(action.htmlText).toEqual("<a href=\"https://matrix.to/#/@jc.denton:unatco.gov\">JCDenton</a>, it's a bomb!");
+    });
+    it("should highlight a user, regardless of case", () => {
+        let action = new MatrixAction("message", "JCDenton, it's a bomb!", "JCDenton, it's a bomb!", null);
+        action.formatMentions({
+            "jcdenton": "@jc.denton:unatco.gov"
+        });
+        expect(action.htmlText).toEqual("<a href=\"https://matrix.to/#/@jc.denton:unatco.gov\">jcdenton</a>, it's a bomb!");
+    });
+    it("should highlight a user, with plain text", () => {
+        let action = new MatrixAction("message", "JCDenton, it's a bomb!");
+        action.formatMentions({
+            "JCDenton": "@jc.denton:unatco.gov"
+        });
+        expect(action.htmlText).toEqual("<a href=\"https://matrix.to/#/@jc.denton:unatco.gov\">JCDenton</a>, it's a bomb!");
+    });
+    it("should highlight a user, with weird characters", () => {
+        let action = new MatrixAction("message", "JCDenton[m], it's a bomb!");
```

Comment:
just for fun, should make the nick weirder: `JCDenton|$m|` or something, plus a backtick at the beginning or end of the nick.